jolt-core/src/poly/structured_poly.rs

```diff
@@ -20,7 +22,9 @@ pub trait BatchablePolynomials {
     /// uses `DensePolynomial::merge` to combine polynomials of the same size.
     fn batch(&self) -> Self::BatchedPolynomials;
     /// Commits to batched polynomials, typically using `DensePolynomial::combined_commit`.
-    fn commit(batched_polys: &Self::BatchedPolynomials) -> Self::Commitment;
+    fn commit(batched_polys: &Self::BatchedPolynomials, initalizer: &PedersenInit<G>) -> Self::Commitment;
+
+    fn max_generator_size(&self) -> usize;
```

Comment:
I think the implementations will be cleaner if we change the signature to:
```suggestion
    fn max_generator_size(&Self::BatchedPolynomials) -> usize;
```

Reply:
I did it this way so we could theoretically split the call to `batch()` and the call to `max_generator_size()`.

Reply:
It's not too bad, lmk what you think #184